pkg/utils/k8sutils.go

```diff
@@ -41,6 +41,9 @@ const StatusReporterImageEnvVar = "STATUS_REPORTER_IMAGE"
 // Value corresponding to annotation key has subscription channel
 const DesiredSubscriptionChannelAnnotationKey = "ocs.openshift.io/subscription.channel"
 
+// Value corresponding to annotation key has desired client hash
+const DesiredConfigHashAnnotationKey = "ocs.openshift.io/desired.config.hash"
```

Comment:
1. This is the name of an annotation on the CR so by default it represents the desired state. There is no reason to mention desired in the name of the annotation. 
2. The encoding of the data `hash` might change in the future, I don't think we need to mention that as part of the name as it is an impl detail.
3. The annotation name does not describe what kind of function this annotation represents

Reply:
how about changing it to `consumer.config.state`

Reply:
What about `ocs.openshift.io/provider-side-state`? or something like that?

Reply:
just `ocs.openshift.io/provider-state` is good enough i guess

Reply:
`provider-state` implies this represents the entire provider state which is wrong. 
`provider-side-state` implies this represent some state on the provider side